independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1089,6 +1091,46 @@ private void implementCreateForSyntheticBean(ClassCreator beanCreator, BeanInfo
                 .setModifiers(ACC_PRIVATE);
         bean.getCreatorConsumer().accept(createSynthetic);
 
+        // TODO move below, after `SyntheticCreationalContextImpl` is created, and pass it to this method
+        Consumer<MethodCreator> isActive = bean.getIsActiveConsumer();
+        if (isActive != null) {
+            MethodCreator isActiveMethod = beanCreator.getMethodCreator("isActiveInternal", ActiveResult.class)
+                    .setModifiers(ACC_PRIVATE);
+            isActive.accept(isActiveMethod);
+
+            List<InjectionPointInfo> matchingIPs = new ArrayList<>();
+            for (InjectionPointInfo injectionPoint : bean.getDeployment().getInjectionPoints()) {
+                if (bean.equals(injectionPoint.getResolvedBean())) {
+                    matchingIPs.add(injectionPoint);
+                }
+            }
+
+            ResultHandle activeResult = doCreate.invokeSpecialMethod(isActiveMethod.getMethodDescriptor(), doCreate.getThis());
+            ResultHandle activeResultBool = doCreate.invokeVirtualMethod(
+                    MethodDescriptor.ofMethod(ActiveResult.class, "isActive", boolean.class), activeResult);
+            BytecodeCreator notActive = doCreate.ifFalse(activeResultBool).trueBranch();
+            StringBuilderGenerator msg = Gizmo.newStringBuilder(notActive);
+            msg.append("Bean is not active: ");
+            msg.append(Gizmo.toString(notActive, notActive.getThis()));
+            msg.append("\nReason: ");
+            msg.append(notActive.invokeVirtualMethod(
+                    MethodDescriptor.ofMethod(ActiveResult.class, "inactiveReason", String.class), activeResult));
+            if (!matchingIPs.isEmpty()) {
+                msg.append("\nThis bean is injected into:");
```

Comment:
Could we add the following hint, so that providers of inactive beans (Agroal, reactive clients, Flyway, Liquibase, ...) don't need to add it in their own messages?

See my other message for a more detailed motivation.

```suggestion
            ResultHandle typeNameString = /* ... TODO ... */;
            msg.append("\nTo keep this bean inactive and avoid this exception:");
            msg.append("\n\t- Configure all extensions consuming this bean as inactive as well, if they allow it, e.g. 'quarkus.someextension.active = false'");
            msg.append("\n\t- Make sure that custom code accessing this bean only does so if the bean is active");
            if (!matchingIPs.isEmpty()) {
                msg.append("\n\t- Inject custom CDI beans with 'Instance<").append(typeNameString).append(">' instead of '").append(typeNameString).append("'");
                msg.append("\nThis bean is injected into:");
```

Reply:
Makes sense, I'll do that.